### ISSUE TITLE
Missing support for Steam's "Game File Transfer over Local Network"

### ISSUE THREAD

**Issue by StefanBruens** [firewalld/firewalld#1163]
Current versions of the Steam Client support downloading game files from a device in the local network, instead of downloading everything from the remote Steam servers.

For this to work, the device supplying the files must have TCP port 27040 opened.

**Comment by StefanBruens** [comment]
[steam-filetransfer.xml.txt](https://github.com/firewalld/firewalld/files/12008578/steam-filetransfer.xml.txt)

**Comment by devurandom** [comment]
> [steam-filetransfer.xml.txt](https://github.com/firewalld/firewalld/files/12008578/steam-filetransfer.xml.txt)

Confirmed: Loading this definition into firewalld on the *receiving* side (I already had a manually created `nft` rule on the sending side, i.e. the machine that already had the files) allows file transfers over the local network.

```
# firewall-cmd --permanent --new-service-from-file=steam-filetransfer.xml
# firewall-cmd --permanent --add-service=steam-filetransfer
# firewall-cmd --reload
```

**Comment by erig0** [comment]
The steam support forums [1] say:

> The network transfer happens on TCP port 27040 and it needs to be allowed by local firewall software. You also need to open UDP ports 27031-27036 for client discovery.

We can definitely add a service with `27040/tcp`. The question is, do we need to also allow `27031-27036/udp`?

[1]: https://help.steampowered.com/en/faqs/view/46BD-6BA8-B012-CE43

**Comment by devurandom** [comment]
Yes, according to the docs those are also needed for discovery. But they are also included in https://github.com/firewalld/firewalld/blob/385fbbf0edc2626a812385abd15c965a3714eff4/config/services/steam-streaming.xml. Would that be a problem, two services containing the same ports?

**Comment by erig0** [comment]
No. Not a problem.

**Comment by devurandom** [comment]
> The steam support forums [1](https://help.steampowered.com/en/faqs/view/46BD-6BA8-B012-CE43) say:
> 
> > The network transfer happens on TCP port 27040 and it needs to be allowed by local firewall software. You also need to open UDP ports 27031-27036 for client discovery.
> 
> We can definitely add a service with `27040/tcp`. The question is, do we need to also allow `27031-27036/udp`?

@koloved [tested this](https://github.com/firewalld/firewalld/pull/1216#issuecomment-2143409846): Yes, these ports are also needed.

**Comment by erig0** [comment]
Fixed via #1216
